File: Patient.R
# 1
install.packages('survminer')
install.packages("survival")
library(survminer)
library(survival)
# Importing the dataset
Patient <- read.csv('C:\\Users\\WIN10\\Desktop\\LEARNING\\DS\\survival\\Patient.csv')
attach(Patient)
str(Patient)
# Define variables 
time <- Followup
event <- Eventtype
group <-Scenario  # unemployment insurance can take 2 values 0 or 1 
# Descriptive statistics
summary(time)
table(event)
table(group)
# Kaplan-Meier non-parametric analysis
kmsurvival <- survfit(Surv(time, event) ~ 1)
summary(kmsurvival)
plot(kmsurvival, xlab="Time", ylab="Survival Probability")
ggsurvplot(kmsurvival, data=Patient, risk.table = TRUE)

# Kaplan-Meier non-parametric analysis by group
kmsurvival1 <- survfit(Surv(time, event) ~ group)
summary(kmsurvival1)
plot(kmsurvival1, xlab="Time", ylab="Survival Probability")
ggsurvplot(kmsurvival1, data=Patient, risk.table = TRUE)

File: Patient.py
# 1
pip install lifelines
import lifelines
import pandas as pd
# importing the dataset
patient= pd.read_csv("C:\\Users\\WIN10\\Desktop\\LEARNING\\DS\\survival\\Patient.csv")
# Performing EDA
patient.head()
patient.describe()
patient.drop(['PatientID'], axis =1, inplace =True) # Dropping ID column
# Creating the dummy variables
patient = pd.get_dummies(patient)
patient["Followup"].describe()
# Spell is referring to time 
T = patient.Followup

# Importing the KaplanMeierFitter model to fit the survival analysis
from lifelines import KaplanMeierFitter
# Initiating the KaplanMeierFitter model
kmf = KaplanMeierFitter()
# Fitting KaplanMeierFitter model on Time and Events for death 
kmf.fit(T, event_observed=patient.Scenario_A)
# Time-line estimations plot 
kmf.plot()

# Over Multiple groups 
# For each group, here group is ui
patient.Scenario_A.value_counts()
# Applying KaplanMeierFitter model on Time and Events for the group "1"
kmf.fit(T[patient.Eventtype==1], patient.Eventtype[patient.Eventtype==1], label='1')
ax = kmf.plot()
# Applying KaplanMeierFitter model on Time and Events for the group "0"
kmf.fit(T[patient.Eventtype==0], patient.Eventtype[patient.Eventtype==0], label='0')
ax = kmf.plot()


